src/cfdp_core/pdu_directive.cpp

```diff
@@ -111,3 +113,107 @@ std::vector<uint8_t> cfdp::pdu::directive::AckPdu::encodeToBytes() const
+                                              T fileSize, uint8_t lengthOfEntityID,
+                                              uint64_t faultEntityID)
+    : conditionCode(conditionCode), checksum(checksum), fileSize(fileSize),
+      lengthOfEntityID(lengthOfEntityID), faultEntityID(faultEntityID)
+{
+    if (not isError())
+    {
+        throw exception::PduConstructionException(
+            "Fault location should be omitted with `No error` condition code");
+    }
+};
+
+template <class T>
+cfdp::pdu::directive::EndOfFile<T>::EndOfFile(std::span<uint8_t const> memory,
+                                              uint8_t lengthOfEntityID)
+    : lengthOfEntityID(lengthOfEntityID)
+{
+    const auto memory_size = memory.size();
+
+    if (memory_size < const_pdu_size_bytes)
+    {
+        throw exception::DecodeFromBytesException("Passed memory does not contain enough bytes");
+    }
+
+    if (memory[0] != utils::toUnderlying(Directive::Eof))
+    {
+        throw exception::DecodeFromBytesException("File Directive code is not End of File Pdu");
+    }
+    const auto secondByte = memory[1];
+
+    conditionCode = Condition((secondByte & eof_condition_code_bitmask) >> 4);
+    checksum      = utils::bytesToInt<uint64_t>(memory, 2, sizeof(uint32_t));
+    fileSize      = utils::bytesToInt<uint64_t>(memory, 6, getSizeOfFileSize());
+
+    if (not isError())
+    {
+        return;
+    }
+
+    if (memory_size != const_pdu_size_bytes + getFaultLocationSize())
+    {
+        throw exception::DecodeFromBytesException("Passed memory has invalid size");
+    }
+
+    const auto fault_location_position = 6 + getSizeOfFileSize();
+
+    if (memory[fault_location_position] != utils::toUnderlying(TLVType::EntityId))
+    {
+        throw exception::DecodeFromBytesException("TLVType is not Enitity Id");
+    }
+    faultEntityID =
+        utils::bytesToInt<uint64_t>(memory, fault_location_position + 1, lengthOfEntityID);
+};
+
+template <class T>
+std::vector<uint8_t> cfdp::pdu::directive::EndOfFile<T>::encodeToBytes() const
+{
+    const auto pdu_size = getRawSize();
+    auto encodedPdu     = std::vector<uint8_t>{};
+
+    encodedPdu.reserve(pdu_size);
+
+    encodedPdu.push_back(utils::toUnderlying(Directive::Eof));
+
+    const uint8_t secondByte = (utils::toUnderlying(conditionCode) << 4);
+
+    encodedPdu.push_back(secondByte);
+
+    auto checksumBytes = utils::intToBytes(checksum, sizeof(uint32_t));
+    utils::concatenateVectorsInplace(checksumBytes, encodedPdu);
+
+    auto fileSizeBytes = utils::intToBytes(fileSize, getSizeOfFileSize());
+    utils::concatenateVectorsInplace(fileSizeBytes, encodedPdu);
+
+    if (not isError())
+    {
+        return encodedPdu;
+    }
+
+    encodedPdu.push_back(utils::toUnderlying(TLVType::EntityId));
+
+    auto faultEntityIDBytes = utils::intToBytes(faultEntityID, lengthOfEntityID);
+    utils::concatenateVectorsInplace(faultEntityIDBytes, encodedPdu);
```

Comment:
TLV format is wrong
![image](https://github.com/user-attachments/assets/643128a3-1097-4ea3-8b34-9ff0789f65ac)

Reply:
fixed